app/server.php

```diff
@@ -83,6 +84,77 @@ function($request, $response) {
         }
     );
 
+App::get('/todos')
+    ->inject('request')
+    ->inject('response')
+    ->action(
+        function($request, $response) {
+            $path = \realpath('/app/app/todos.json');
+            $data = json_decode(file_get_contents($path), true);
+            $response->json($data);
+        }
+    );
+
+App::post('/todos')
+    ->inject('response')
+    ->param('task', "", new Wildcard(), 'Prefs key-value JSON object.')
+    ->param('is_complete', true, new Wildcard(), 'Tells whether task is complete or not')
+    ->action(
+        function($task, $is_complete, $response) {
+            $path = \realpath('/app/app/todos.json');
+            $data = json_decode(file_get_contents($path));
+            $task_entry = ['task' => $task, 'is_complete' => $is_complete];
+            array_push($data, $task_entry);
+            $jsonData = json_encode($data, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+            file_put_contents($path, $jsonData);
+            $response->json($jsonData);
+        }
+    );
+
+App::put('/todos/:todo_name')
+    ->inject('response')
+    ->param('todo_name', "", new Wildcard(), 'name of the todo')
+    ->param('is_complete', true, new Wildcard(), 'Tells whether task is complete or not')
+    ->action(
+        function($todo_name, $is_complete, $response) {
+            $path = \realpath('/app/app/todos.json');
+            $data = json_decode(file_get_contents($path));
+            // var_dump($data);
+            foreach($data as $object){
+                if($object->task == $todo_name){
+                    $object->is_complete = $is_complete;
+                    break;
+                }
+            }
+            // var_dump($data);
+            $jsonData = json_encode($data, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+            file_put_contents($path, $jsonData);
+            $response->json($data);
+        }
+    );
+
+App::delete('/todos/:todo_name')
+    ->inject('response')
+    ->param('todo_name', "", new Wildcard(), 'name of the todo')
+    ->action(
+        function($todo_name, $response) {
+            $path = \realpath('/app/app/todos.json');
+            $data = json_decode(file_get_contents($path));
+            var_dump($todo_name);
+            foreach($data as $object => $item){
+                if($item->task == $todo_name){
+                    var_dump($object);
+                    unset($data[$object]);
+                }
+            }
+            var_dump($data);
+            $jsonData = json_encode($data, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+            file_put_contents($path, $jsonData);
+            $response->json($data);
```

Comment:
I think it's a convention with DELETE endpoints , to return the item that was deleted. Can we do that ?

Reply:
There's no hard rule here - I believe the following are appropriate:
- Respond with 204 and an empty response body (what we do in Appwrite)
- Respond with a 200 and the item that was deleted

[ref](https://stackoverflow.com/questions/25970523/restful-what-should-a-delete-response-body-contain)

Reply:
@vermakhushboo  let's follow the appwrite convention in that case 🙂